extensions/KHR/SPV_KHR_cooperative_matrix.asciidoc

```diff
@@ -268,6 +268,10 @@ where all invocations in 'Scope' cooperate to compute and store the matrix. +
 'Use' must be a 'constant instruction' scalar 32-bit 'integer type' whose
 value corresponds to a 'Cooperative Matrix Use'. +
  +
+All dynamic instances of an instruction with an operand or result that is an
+object of this type must be executed such that all the invocations in the
+'Scope' instance are active or none of them are.
```

Comment:
I thought the consensus was to only require this for the matmul instruction?

Reply:
It was also the case for the load/store instructions. If we make this proposed change, it would, strictly speaking, lead to new restrictions (that arguably might have been assumed by some). For example, some of the conversion instructions will result in cross-lane data exchanges for some vendors and uniform control flow is helpful as a restriction. Arguably this restriction might not be required on all instructions but I'm not sure we lose much with it (notably, AFAICT, element-wise access still works according to the letter of this rule). Maybe best discussed in the next teleconference (or F2F)?

Reply:
Discussed at the 2023/10/25 F2F: the proposed change is fine but we should capture the fact that element access needn't be performed in uniform control flow in an issue. The reason for that, i.e. that the uniform control flow requirement as proposed applies to instructions with operands that have a cooperative matrix type but not a pointer to cooperative matrix type, is subtle and easily missed.